source/clear-linux/tutorials/kubernetes.rst

```diff
@@ -0,0 +1,299 @@
+  .. _kubernetes:
+
+Run Kubernetes\* on Clear Linux
+###############################
+
+This tutorial describes how to install, configure, and run the 
+`Kubernetes system`_ on |CL-ATTR| using either the Kubernetes
+runtime interface *CRI-O* and *kata-runtime* or *Docker\** and *runc*.
+
+Kubernetes\* is an open source system for automating deployment, scaling, and
+management of containerized applications. It groups containers that make up an
+application into logical units for easy management and discovery.
+
+Runc and kata-runtime are compatible with :abbr:`OCI (Open Container
+Initiative)` guidelines and work seamlessly with Kubernetes. `Kata
+Containers`_\* provide strong isolation for untrusted workloads or
+multi-tenant scenarios.  Runc and Kata Containers can be allocated on a
+per-container basis so you can mix and match both on the same host to suit
+your needs.
+
+Prerequisites
+*************
+
+This tutorial assumes you have installed |CL| version 25040 or newer on your
+host system.
+For detailed instructions on installing |CL| on a bare metal system, follow
+the :ref:`bare metal installation tutorial<bare-metal-install>`.
+
+Before you install any new packages, update |CL| with the following command:
+
+.. code-block:: bash
+
+   sudo swupd update
+
+Install Kubernetes and CRI runtimes
+***********************************
+
+Kubernetes and a set of supported :abbr:`CRI (Container Runtime Interface)` 
+runtimes are included in the `cloud-native-basic`_ bundle. To install the 
+framework, enter the following command:
+
+.. code-block:: bash
+
+   sudo swupd bundle-add cloud-native-basic
+
+Configure Kubernetes
+********************
+
+This tutorial uses the basic default Kubernetes configuration for simplicity.
+You must define your Kubernetes configuration according to your specific
+deployment and your security needs.
+
+#. Enable IP forwarding:
+
+   Create (or edit if it exists) the file :file:`/etc/sysctl.d/60-k8s.conf`
+   and include the following line:
+
+   .. code-block:: bash
+
+      net.ipv4.ip_forward = 1
+
+   Apply the change:
+
+   .. code-block:: bash
+
+      sudo systemctl restart systemd-sysctl
+
+#. Enable kubelet service:
+
+   .. code-block:: bash
+
+      sudo systemctl enable kubelet.service
+
+#. Disable swap using one of the following methods, either:
+
+   a) Temporarily:
+
+      .. code-block:: bash
+
+         sudo swapoff -a
+
+      .. note::
+
+         Swap will be enabled at next reboot, causing failures in
+         your cluster.
+
+   or:
+
+   b) Permanently:
+
+      Mask the swap partition:
+
+      .. code-block:: bash
+
+         sudo systemctl mask `sed -n -e 's#^/dev/\([0-9a-z]*\).*#dev-\1.swap#p' /proc/swaps
+         sudo swapoff -a
+
+      .. note::
+
+         On systems with limited resources, some performance degradation may
+         be observed while swap is disabled.
+
+#. Configure the Kubernetes runtime interface, either:
```

Comment:
The way this is described makes it seem like you must use runc with dockershium or kata with CRIO.  

While its nice to configure / show both, I think I would clarify the options:
1) dockershim + runc
2) CRIO, where you can use multiple runtimes, including runc and kata-runtime

CRIO (or containerd) provide the most flexibility, and this is getting lost in the tutorial imo.

Reply:
makes sense. kubernetes recently added a doc for "CRI selection/configuration" and we should link to that here once that's published